File: src/main/java/com/github/sorusclient/client/adapter/IAdapter.java
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter;

public interface IAdapter {
    ScreenType getOpenScreen();
    void openScreen(ScreenType screenType);
    double[] getScreenDimensions();
    double[] getMouseLocation();

    IPlayerEntity getPlayer();
    IWorld getWorld();

    PerspectiveMode getPerspective();
    void setPerspective(PerspectiveMode perspectiveMode);

    IServer getCurrentServer();

    IKeyBind getKeyBind(KeyBindType keyBindType);

    void sendPlayerMessage(String message);

    void setDisplayTitle(String title);
    void setDisplayIcon(String iconSmall, String iconLarge);

    void leaveWorld();
    void joinServer(String ip);

    GameMode getGameMode();

    ISession getSession();

    IText createText(String string);

    String getVersion();

    IRenderer getRenderer();

    int getFPS();

}

File: src/main/kotlin/com/github/sorusclient/client/adapter/IAdapter.kt
/*
 * Copyright (c) 2022. Danterus
 * Copyright (c) 2022. Sorus Contributors
 *
 * SPDX-License-Identifier: MPL-2.0
 */

package com.github.sorusclient.client.adapter

import com.github.sorusclient.client.adapter.IKeyBind.KeyBindType

interface IAdapter {
    fun openScreen(screenType: ScreenType)
    val openScreen: ScreenType?

    val screenDimensions: DoubleArray
    val mouseLocation: DoubleArray

    val player: IPlayerEntity?
    val world: IWorld
    var perspective: PerspectiveMode
    val currentServer: IServer?
    val gameMode: GameMode
    val session: ISession
    val fps: Int

    fun getKeyBind(type: KeyBindType): IKeyBind
    fun sendPlayerMessage(message: String)
    fun createText(string: String): IText

    fun setDisplayTitle(title: String)
    fun setDisplayIcon(iconSmall: String, iconLarge: String)

    fun leaveWorld()
    fun joinServer(ip: String)

    val version: String

    val renderer: IRenderer
}